extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1627,11 +1664,15 @@ private void checkPendingCompletionTasks() throws ExecutionException, Interrupte
           continue;
         }
 
-        Iterator<Map.Entry<String, TaskData>> iTask = group.tasks.entrySet().iterator();
+        Iterator<Entry<String, TaskData>> iTask = group.tasks.entrySet().iterator();
         while (iTask.hasNext()) {
-          Map.Entry<String, TaskData> task = iTask.next();
+          final Entry<String, TaskData> entry = iTask.next();
+          final String taskId = entry.getKey();
+          final TaskData taskData = entry.getValue();
+
+          Preconditions.checkNotNull(taskData.status, "task[%s] has a null status", taskId);
```

Comment:
When will this get thrown and what will happen when it gets thrown? I'm wondering what the user experience is going to be like.

Reply:
This should never happen. NPE would throw if `taskData.status` is null and this is just a sanity code to see what's null on the potential NPE. I improved the error message.

Reply:
Thanks.